zwavegeneric/integrationpluginzwavegeneric.cpp

```diff
@@ -0,0 +1,203 @@
+        info->finish(Thing::ThingErrorSetupFailed);
+        return;
+    }
+
+    if (thing->thingClassId() == powerSocketThingClassId) {
+
+        thing->setStateValue(powerSocketPowerStateTypeId, node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassSwitchBinary, 1, 0, ZWaveValue::TypeBool).value().toBool());
+        info->finish(Thing::ThingErrorNoError);
+
+        connect(node, &ZWaveNode::valueChanged, thing, [thing](const ZWaveValue &value){
+            if (value.genre() == ZWaveValue::GenreUser && value.commandClass() == ZWaveValue::CommandClassSwitchBinary) {
+                thing->setStateValue(powerSocketPowerStateTypeId, value.value().toBool());
+            }
+        });
+    }
+
+    if (thing->thingClassId() == powerMeterSocketThingClassId) {
+
+        thing->setStateValue(powerMeterSocketPowerStateTypeId, node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassSwitchBinary, 1, 0, ZWaveValue::TypeBool).value().toBool());
+        info->finish(Thing::ThingErrorNoError);
+
+        thing->setStateValue(powerMeterSocketCurrentPowerStateTypeId, node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassSensorMultilevel, 1, 4, ZWaveValue::TypeDecimal).value().toDouble());
+        thing->setStateValue(powerMeterSocketTotalEnergyConsumedStateTypeId, node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassMeter, 1, 0, ZWaveValue::TypeDecimal).value().toDouble());
+        connect(node, &ZWaveNode::valueChanged, thing, [thing](const ZWaveValue &value){
+            if (value.genre() == ZWaveValue::GenreUser && value.commandClass() == ZWaveValue::CommandClassSwitchBinary) {
+                thing->setStateValue(powerMeterSocketPowerStateTypeId, value.value().toBool());
+            }
+            if (value.genre() == ZWaveValue::GenreUser && value.commandClass() == ZWaveValue::CommandClassSensorMultilevel) {
+                thing->setStateValue(powerMeterSocketCurrentPowerStateTypeId, value.value().toDouble());
+            } if (value.genre() == ZWaveValue::GenreUser && value.commandClass() == ZWaveValue::CommandClassMeter && value.index() == 0) {
+                thing->setStateValue(powerMeterSocketTotalEnergyConsumedStateTypeId, value.value().toDouble());
+            }
+        });
+    }
+
+    if (thing->thingClassId() == doorLockThingClassId) {
+
+        thing->setStateValue(doorLockStateStateTypeId, node->value(ZWaveValue::GenreUser, ZWaveValue::CommandClassDoorLock, 1, 0, ZWaveValue::TypeBool).value().toBool() ? "locked" : "unlocked");
+        info->finish(Thing::ThingErrorNoError);
+
+        connect(node, &ZWaveNode::valueChanged, thing, [thing](const ZWaveValue &value){
+            if (value.genre() == ZWaveValue::GenreUser && value.commandClass() == ZWaveValue::CommandClassDoorLock && value.index() == 0) {
+                thing->setStateValue(doorLockStateStateTypeId, value.value().toBool() ? "locked" : "unlocked");
+            }
+        });
+    }
+
+    info->finish(Thing::ThingErrorNoError);
```

Comment:
if you don't return at the end of each thing class `if`, you finish the info multiple times.